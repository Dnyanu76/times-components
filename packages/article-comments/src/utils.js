const storeURL = {
  gb:
    "https://thetimes.co.uk/subscribe/digital?ILC=GB-TNL_The_Times-Conversion_Page-Homepage-2020",
  ie:
    "https://store.thetimes.ie/?ILC=IE-TNL_The_Times-Conversion_Page-Homepage-2020",
  global:
    "https://globalstore.thetimes.co.uk/?ILC=INTL-TNL_The_Times-Conversion_Page-Homepage-2020"
};

export const userShouldUpdateName = async username => {
  if (!username) {
    return false;
  }
  const url = `/api/comments/display-names-pseudonyms?username=${username}`;

  const checkUsername = global
    .fetch(url)
    .then(response => response.json())
    .then(data => data);

  const isPseudonym = await checkUsername;

  if (!isPseudonym) {
    return false;
  }

  const bannerCount = global.window.localStorage.getItem(
    "realNameCommentingBannerViewCount"
  );
  const isBannerVisible = global.window.localStorage.getItem(
    "isRealNameCommentingBannerVisible"
  );
  const hasLocalStorageBeenSet = bannerCount && isBannerVisible;

  if (!hasLocalStorageBeenSet) {
    global.window.localStorage.setItem("realNameCommentingBannerViewCount", 3);
    global.window.localStorage.setItem(
      "isRealNameCommentingBannerVisible",
      false
    );
  }

  return bannerCount > 0;
};

export default () => {
  const region =
    // eslint-disable-next-line no-undef
    typeof window !== "undefined" && window.nuk && window.nuk.region;
  return storeURL[region] || storeURL.gb;
};

const parseCookie = (value, valueSeparator, paramSeparator) => {
  console.log("***value parse cookie***", value);
  value = decodeURIComponent(value)

  return value.split(valueSeparator).reduce((res, param) => {
      const parts = param.split(paramSeparator);
      res[parts[0]] = parts[1];
      return res;
  }, {});
};
  const getCpnId = value => {
    console.log("***value***", value);
    const cookie = parseCookie(value, '&', '=');
    console.log("***cookie ***", cookie );
    return cookie.eid || null;
};

export const reauthenticateUser = () => {
  const spotImAccessToken = global.window.localStorage.getItem('SPOTIM_ACCESS_TOKEN');
  const spotImDeviceUuid = global.window.localStorage.getItem('SPOTIM_DEVICE_UUID_V2');
  const spotAB = global.window.localStorage.getItem('SPOT_AB');
  const spotImDeviceV2 = global.window.localStorage.getItem('SPOTIM_DEVICE_V2');
  const spotImCurrentUser = global.window.localStorage.getItem('SPOTIM_CURRENT_USER');
  console.log("***1***");
  if (!spotImAccessToken) {
    console.log("***2***");
    return
  }
  console.log("***3***");
  // u_rAGPG2gW1OUB
  // u_sE9UoUYxWQ50
    const acsCookie = window.nuk.getCookieValue('acs_tnl')
    console.log("***acsCookie***", acsCookie);
    const cpn = getCpnId(acsCookie);
    console.log("***cpn***", cpn);
//?enableRealNameCommenting
};

document.addEventListener(
  "spot-im-user-auth-success", (event) => {

    const {displayName, email, id, username} = event.detail


    console.log('XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX')
    console.log('XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX')
    console.log('XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX', event.detail)
    console.log('XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX')
    console.log('XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX')
  },
);
