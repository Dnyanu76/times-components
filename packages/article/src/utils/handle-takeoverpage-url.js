import React, { useState } from "react";
import { WebView } from "react-native-webview";
import { ActivityIndicator } from "react-native";

export default url => {
  const [isSpinnerShown, hideSpinner] = useState(true);
  return (
    <>
      <WebView source={{ uri: url }} onLoad={() => hideSpinner(false)} />
      {isSpinnerShown && <ActivityIndicator size="large" />}
    </>
  );
};
