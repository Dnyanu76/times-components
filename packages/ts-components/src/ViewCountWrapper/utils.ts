export const hasCookieConsent = () =>
  typeof window !== 'undefined' &&
  window.document.cookie.indexOf('nuk-consent-personalisation=1') >= 0;

export const getStorageProvider = () =>
  typeof window !== 'undefined' && window.sessionStorage;
