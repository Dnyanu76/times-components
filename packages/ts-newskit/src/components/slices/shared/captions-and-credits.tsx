import { TextBlock } from 'newskit';
import React from 'react';
import { StyledSpan } from '../shared-styles';
import { ImageProps } from '../lead-article/index';

export const CaptionsAndCredits = ({
  images,
  hasCaption
}: {
  images?: ImageProps;
  hasCaption?: boolean;
}) => {
  if (!images) {
    return null;
  }
  return (
    <TextBlock
      marginBlockStart="space020"
      stylePreset="inkSubtle"
      typographyPreset="editorialCaption010"
    >
      {images.caption}
      <StyledSpan hasCaption={hasCaption}>{images.credits}</StyledSpan>
    </TextBlock>
  );
};
