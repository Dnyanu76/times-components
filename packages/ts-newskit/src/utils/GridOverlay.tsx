import { Block, Cell, Grid, styled } from 'newskit'
import React from 'react'

const StyledGrid = styled(Grid)<{show?: boolean}>`
  display: ${({show}) => show ? 'flex' : 'none'};
  position: fixed;
  height: 100vh;
  width: 100%;
  text-align: center;
  top: 0;
  z-index: 1;
`
const StyledCellBlock = styled(Block)`
  background-color: rgba(0,0,0,0.1);
  height: 100%;
  width: 100%;
`

export const GridOverlay = ({ show }: { show?: boolean }) => {
  const cellsArray = [...Array(12).keys()]
  const cells = cellsArray.map(cell => (
    <Cell>
      <StyledCellBlock>{cell + 1}</StyledCellBlock>
    </Cell>
  ))

  return (
    <StyledGrid xsMargin="space045" mdMargin="space050" show={show}>
      {cells}
    </StyledGrid>
  )
}
