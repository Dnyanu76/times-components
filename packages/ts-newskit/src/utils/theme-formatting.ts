import { Theme } from "newskit";

const modifyBaseRemValue = (multiplier: number, valueToUpdate: string) => {
  const getVal = valueToUpdate.slice(0, -3);
  const updatedVal = multiplier * Number(getVal);
  
  return `${updatedVal}rem`
}

const modifyFontFamilyValue = (family: string, weight?: number) => {
  let updatedFont = family;
  switch (family) {
    case "Roboto":
      updatedFont = weight && weight > 500 ? "Roboto-Medium" : "Roboto-Regular"
      break;
    case "Times Modern":
      updatedFont = weight && weight > 500 ? "TimesModern-Bold" : "TimesModern-Regular"
      break;
    case "Times Digital W04 Regular":
      updatedFont = "TimesDigitalW04-Regular"
      break;
  
    default:
      family
      break;
  }

  return updatedFont;
}

  
export const updateThemeTypography = (themeObj: Theme) => {
  // Update Borders
  themeObj.borders = Object.fromEntries(
    Object.entries(themeObj.borders).map(border => {
      const updatedVal = typeof(border[1]) === 'string' && border[1].includes("rem")
        ? modifyBaseRemValue(1.6, border[1])
        : border[1]
      
      return [border[0], updatedVal]
    })
  );

  // Update Font Sizes
  themeObj.fonts = Object.fromEntries(
    Object.entries(themeObj.fonts).map(font => {
      const updatedVal = typeof(font[1]) === 'string' && font[1].includes("rem")
      ? modifyBaseRemValue(1.6, font[1])
      : font[1]
      
      return [font[0], updatedVal]
    })
  );

  // Update Font Families
  Object.values(themeObj.fonts).map(font => {
    if(font.fontFamily) {
      font.fontFamily = font.fontFamily
        ? modifyFontFamilyValue(font.fontFamily)
        : null      
    }
  });  

  // Update Typography Presets
  Object.values(themeObj.typographyPresets).map(preset => {
    preset.fontFamily = modifyFontFamilyValue(String(preset.fontFamily), Number(preset.fontWeight));
    preset.fontWeight = 400;

    preset.fontSize = typeof(preset.fontSize) === 'string' && preset.fontSize.includes("rem")
      ? modifyBaseRemValue(1.6, preset.fontSize)
      : preset.fontSize;
  });  
  
  return themeObj;
}
  
export const formatThemeOverrides = (themeObj: Theme, stylePresets = {}, typographyPresets = {}) => {
  const updatedThemeTypography = updateThemeTypography(themeObj);
  console.log("updatedThemeTypography: ", updatedThemeTypography)

  return {
    overrides: {
      ...updatedThemeTypography,
      breakpoints: {
        xs: 0,
        sm: 520,
        md: 768,
        lg: 1024,
        xl: 1320
      },
      typographyPresets: {
        ...updatedThemeTypography.typographyPresets,
        ...typographyPresets
      },
      stylePresets
    }  
  }
}