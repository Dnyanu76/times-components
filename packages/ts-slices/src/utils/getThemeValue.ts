export const getThemeValue = (theme: Record<string, string>, value: string) =>
  theme[value];
