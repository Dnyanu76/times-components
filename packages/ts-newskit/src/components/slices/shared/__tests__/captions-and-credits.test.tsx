import React from 'react';
import { render } from '../../../../utils/test-utils';
import { CaptionsAndCredits } from '../captions-and-credits';
import '@testing-library/jest-dom';

describe('CaptionsAndCredits', () => {
  it('should render a snapshot', () => {
    const images = {
      alt: '',
      credits: 'Gareth Fuller/Press Association',
      caption: 'Saracens an inclusive club? They didnt look out for me',
      crops: [
        {
          url: 'https://www.example.com/image1.jpg',
          ratio: '3:2'
        }
      ]
    };
    const { asFragment } = render(
      <CaptionsAndCredits images={images} hasCaption={true} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders correctly when images are provided but caption is missing', () => {
    const images = {
      alt: '',
      credits: 'Gareth Fuller/Press Association',
      crops: [
        {
          url: 'https://www.example.com/image1.jpg',
          ratio: '3:2'
        }
      ]
    };

    const { getByText, queryByText } = render(
      <CaptionsAndCredits images={images} hasCaption={true} />
    );
    const captionElement = queryByText(
      /Saracens an inclusive club? They didnt look out for me/i
    );
    const creditsElement = getByText(/Gareth Fuller\/Press Association/i);

    expect(captionElement).toBeNull();
    expect(creditsElement).toBeInTheDocument();
  });

  it('renders correctly when images are provided but hasCaption is false', () => {
    const images = {
      alt: '',
      credits: 'Gareth Fuller/Press Association',
      caption: 'Saracens an inclusive club? They didnt look out for me',
      crops: [
        {
          url: 'https://www.example.com/image1.jpg',
          ratio: '3:2'
        }
      ]
    };

    const { getByText, queryByText } = render(
      <CaptionsAndCredits images={images} hasCaption={false} />
    );

    const captionElement = queryByText(
      /Saracens an inclusive club? They didnt look out for me/i
    );
    const creditsElement = getByText(/Gareth Fuller\/Press Association/i);
    expect(captionElement).toBeNull();
    expect(creditsElement).toBeInTheDocument();
  });

  it('renders nothing when images are not provided', () => {
    const { container } = render(<CaptionsAndCredits />);
    expect(container.firstChild).toBeNull();
  });
});
