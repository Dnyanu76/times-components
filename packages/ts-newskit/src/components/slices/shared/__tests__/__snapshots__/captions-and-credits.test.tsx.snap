// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaptionsAndCredits should render a snapshot 1`] = `
<DocumentFragment>
  <p
    class="css-1c5i4sy"
  >
    Saracens an inclusive club? They didnt look out for me
    <span
      class="css-bbx0sz"
    >
      Gareth Fuller/Press Association
    </span>
  </p>
</DocumentFragment>
`;
